tool/site/client-react/src/components/dispense/createDispenseNotification.ts

```diff
@@ -0,0 +1,265 @@
+import * as fhir from "fhir/r4"
+import {
+  Bundle,
+  CodeableConcept,
+  Extension,
+  Identifier,
+  MedicationDispense,
+  MedicationDispensePerformer,
+  MedicationRequest,
+  MessageHeader,
+  Patient,
+  Quantity,
+  Reference
+} from "fhir/r4"
+import {DispenseFormValues, LineItemFormValues, PrescriptionFormValues} from "./dispenseForm"
+import * as uuid from "uuid"
+import {
+  getLongFormIdExtension,
+  RepeatInformationExtension,
+  TaskBusinessStatusExtension,
+  URL_GROUP_IDENTIFIER_EXTENSION,
+  URL_TASK_BUSINESS_STATUS
+} from "../../fhir/customExtensions"
+import {
+  LineItemStatus,
+  PrescriptionStatus,
+  VALUE_SET_LINE_ITEM_STATUS,
+  VALUE_SET_NON_DISPENSING_REASON,
+  VALUE_SET_PRESCRIPTION_STATUS
+} from "../../fhir/reference-data/valueSets"
+import {getMedicationRequestLineItemId, getRepeatsIssuedAndAllowed} from "../../fhir/helpers"
+
+const EVENT_CODING_DISPENSE_NOTIFICATION = {
+  system: "https://fhir.nhs.uk/CodeSystem/message-event",
+  code: "dispense-notification",
+  display: "Dispense Notification"
+}
+
+export function createDispenseNotification(
+  prescriptionOrderMessageHeader: MessageHeader,
+  patient: Patient,
+  medicationRequests: Array<MedicationRequest>,
+  dispenseFormValues: DispenseFormValues
+): Bundle {
+  const medicationDispenses = medicationRequests.map(medicationRequest => {
+    const lineItemId = getMedicationRequestLineItemId(medicationRequest)
+    const formValuesForLineItem = dispenseFormValues.lineItems.find(item => item.id === lineItemId)
+    return createMedicationDispense(medicationRequest, formValuesForLineItem, dispenseFormValues.prescription)
+  })
+
+  //TODO - Fix resources without IDs in examples? Or just assign a fresh ID here
+  if (!patient.id) {
+    patient.id = medicationRequests[0].subject.reference.substring("urn:uuid:".length)
+  }
+
+  //TODO - work out why we're getting validation errors
+  patient.identifier[0] = {
```

Comment:
what is happening here?

Reply:
Something about nhs number verification status

Reply:
Haven't looked into it yet